File: Algorithms/Recursive Algorithms/C++/Factorial.cpp
#include <bits/stdc++.h> 

using namespace std; 

// get the factorial number(recusrive)
int fact(int n)
{
    if(n==0)
        return 1;    
    else{
        return n*fact(n-1);
    }    
}

int main()
{
    int t; 
    cout << "Enter a number to find factorial: ";
    cin >> t;
    cout << "Factorial of " << t << " is " << fact(t);
    return 0;
}

File: Algorithms/Recursive Algorithms/Go/factorial.go
package main

import "fmt"

func factorial(num int) int {
	if num == 0 || num == 1 {
		return 1
	}
	return num * factorial(num-1)
}

func main() {

	var num int

	fmt.Print("Enter the  number : ")
	fmt.Scanln(&num)
	fmt.Print("Factorial of given number : ")
	fmt.Print(factorial(num))

}
